WalletWasabi/Crypto/ZeroKnowledge/NonInteractive/FiatShamirTransform/Statement.cs

```diff
@@ -0,0 +1,71 @@
+using System.Linq;
+using System.Collections.Generic;
+using WalletWasabi.Helpers;
+
+namespace WalletWasabi.Crypto.ZeroKnowledge.NonInteractive.FiatShamirTransform
+{
+	public class Statement : IStatement
+	{
+		public Statement(LinearRelation.Statement statement)
+		{
+			LinearRelation = statement;
+		}
+
+		// Although in principle this could be an interface type, since only linear
+		// relations are needed for our proofs this is specialized to that type.
+		public LinearRelation.Statement LinearRelation { get; }
+
+		// Plug witness data into this statement. The enumerator should have exactly
```

Comment:
Would it be of any use to add a convenience method:

```csharp
		public IKnowledge ToKnowledge(ScalarVector witness)
		{
			return ToKnowledge(new ScalarVector[] { witness });
		}
```

to make it easier to use the API..?

Reply:
no need as this would not really be used directly, see #4303 for how it would be used but bear in mind that would still not be the main entry point since we will have just a small, fixed set of proofs (e.g. balance proof, range proof, serial number validity proof, etc) and i expect each of these will just have its own constructor or static convenience method